Edit lookup fails - internal server error

For valid page: https://blog.adobe.com/en/2020/12/01/what-does-the-cmo50-tell-us-about-modern-marketing.html#gs.re0ega

Edit URL: https://adobeioruntime.net/api/v1/web/helix/helix-services/content-proxy@v2?owner=adobe&repo=theblog&ref=master&path=%2F&edit=https%3A%2F%2Fblog.adobe.com%2Fen%2F2020%2F12%2F01%2Fwhat-does-the-cmo50-tell-us-about-modern-marketing.html%23gs.re0ega

returns `Internal Server Error`.

Activation logs:

```bash
wsk activation logs fc8055048cc24fc38055048cc22fc32c
2021-01-28T10:17:37.385Z       stdout: instrumenting epsagon.
2021-01-28T10:17:37.386Z       stdout: action-status-begin {"container":{"uuid":"1329520efe5a0d3b619c117839e97724","numInvocations":6,"begin":{"mem":174407680,"age":174227,"concurrency":1}}}
2021-01-28T10:17:37.956Z       stderr: [WARN] StatusCodeError
    at Function.fromError (/nodejsAction/VNX8IJnR/index.js:5135:17)
    at OneDrive.listChildren (/nodejsAction/VNX8IJnR/index.js:4387:37)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  statusCode: 404,
  details: { code: 'itemNotFound', message: 'The resource could not be found.', innerError: { date: '2021-01-28T10:17:37', 'request-id': 'd697e254-021b-480f-9da2-230f67002e79', 'client-request-id': 'd697e254-021b-480f-9da2-230f67002e79' } }
}
2021-01-28T10:17:37.958Z       stderr: [ERROR] error {
  params: {},
  error: StatusCodeError
      at Function.fromError (/nodejsAction/VNX8IJnR/index.js:5135:17)
      at OneDrive.listChildren (/nodejsAction/VNX8IJnR/index.js:4387:37)
      at processTicksAndRejections (internal/process/task_queues.js:97:5) {
    statusCode: 404,
    details: { code: 'itemNotFound', message: 'The resource could not be found.', innerError: { date: '2021-01-28T10:17:37', 'request-id': 'd697e254-021b-480f-9da2-230f67002e79', 'client-request-id': 'd697e254-021b-480f-9da2-230f67002e79' } }
  }
}
2021-01-28T10:17:37.959Z       stderr: error while invoking function TypeError: Cannot read property 'get' of undefined
    at ensureUTF8Charset (/nodejsAction/VNX8IJnR/index.js:579:29)
    at openwhisk (/nodejsAction/VNX8IJnR/index.js:231:5)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /nodejsAction/VNX8IJnR/index.js:716:14
    at async NodeActionRunner.userScriptMain (/nodejsAction/VNX8IJnR/index.js:853:15)
2021-01-28T10:17:37.959Z       stdout: action-status-end {"container":{"uuid":"1329520efe5a0d3b619c117839e97724","numInvocations":6,"begin":{"mem":174407680,"age":174227,"concurrency":1},"end":{"mem":175591424,"age":174800,"concurrency":1},"delta":{"mem":1183744,"age":573}}}
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@adobe/helix-content-proxy",
-  "version": "2.6.3",
+  "version": "2.6.3-test",
   "private": true,
   "description": "Helix Content Proxy",
   "main": "src/index.js",
@@ -45,7 +45,7 @@
   },
   "devDependencies": {
     "@adobe/eslint-config-helix": "1.1.3",
-    "@adobe/helix-deploy": "3.0.3",
+    "@adobe/helix-deploy": "3.0.5",
     "@pollyjs/adapter-node-http": "5.1.0",
     "@pollyjs/core": "5.1.0",
     "@pollyjs/persister-fs": "5.0.0",

--- a/src/lookup-edit-url.js
+++ b/src/lookup-edit-url.js
@@ -52,12 +52,17 @@ async function lookupEditUrl(opts) {
     });
   }
 
-  const location = await handler.getEditUrl({
-    mp,
-    resourcePath,
-    ext,
-    ...opts,
-  });
+  let location;
+  try {
+    location = await handler.getEditUrl({
+      mp,
+      resourcePath,
+      ext,
+      ...opts,
+    });
+  } catch (e) {
+    log.warn(`Handler ${handler.name} threw an error:`, e);
+  }
 
   if (!location) {
     log.error(`Handler ${handler.name} could not lookup ${uri}.`);

--- a/test/onedrive-edit-url.test.js
+++ b/test/onedrive-edit-url.test.js
@@ -21,6 +21,7 @@ const { setupPolly, retrofit } = require('./utils.js');
 
 const fstab = `
 mountpoints:
+  /invalid: https://adobe.sharepoint.com/sites/TheBlog/Shared%20Documents/invalid
   /: https://adobe.sharepoint.com/sites/TheBlog/Shared%20Documents/theblog
 `;
 
@@ -36,7 +37,10 @@ const DEFAULT_PARAMS = {
 };
 
 class FakeOneDrive {
-  getDriveItemFromShareLink() {
+  getDriveItemFromShareLink(url) {
+    if (url === 'https://adobe.sharepoint.com/sites/TheBlog/Shared%20Documents/invalid') {
+      throw new Error('item not found');
+    }
     return {
       id: '01DJQLOW4XEPSY5ROSRRE3SF2L7KJDG7TW',
       webUrl: 'https://adobe.sharepoint.com/sites/TheBlog/Shared%20Documents',
@@ -141,6 +145,21 @@ describe('Onedrive Edit Link Tests', () => {
     const result = await main({
       ...DEFAULT_PARAMS,
       edit: 'https://blog.adobe.com/foo/bar.html',
+    });
+
+    assert.equal(result.statusCode, 404);
+  }).timeout(20000);
+
+  it('Returns 404 for non existent sharelink', async function test() {
+    const { server } = this.polly;
+
+    server
+      .get('https://raw.githubusercontent.com/adobe/theblog/master/fstab.yaml')
+      .intercept((_, res) => res.status(200).send(fstab));
+
+    const result = await main({
+      ...DEFAULT_PARAMS,
+      edit: 'https://blog.adobe.com/invalid/en/2020/11/09/adobe-to-acquire-workfront.html',
     });
 
     assert.equal(result.statusCode, 404);

--- a/test/post-deploy-utils.js
+++ b/test/post-deploy-utils.js
@@ -47,18 +47,6 @@ class OpenwhiskTarget {
   }
 }
 
-class OpenwhiskPreprodTarget extends OpenwhiskTarget {
-  // eslint-disable-next-line class-methods-use-this
-  title() {
-    return 'OpenWhisk - Preprod';
-  }
-
-  // eslint-disable-next-line class-methods-use-this
-  host() {
-    return 'https://preprod.adobeioruntime.net';
-  }
-}
-
 class AWSTarget extends OpenwhiskTarget {
   // eslint-disable-next-line class-methods-use-this
   title() {
@@ -82,7 +70,6 @@ class AWSTarget extends OpenwhiskTarget {
 
 const ALL_TARGETS = [
   OpenwhiskTarget,
-  OpenwhiskPreprodTarget,
   AWSTarget,
 ];
 

--- a/test/post-deploy.js
+++ b/test/post-deploy.js
@@ -68,5 +68,18 @@ createTargets().forEach((target) => {
           throw e;
         });
     }).timeout(20000);
+
+    it('Compute Edit Lookup', async () => {
+      const url = `${target.urlPath()}?owner=adobe&repo=theblog&ref=master&path=%2F&edit=https%3A%2F%2Fblog.adobe.com%2Fen%2F2020%2F12%2F01%2Fwhat-does-the-cmo50-tell-us-about-modern-marketing.html%23gs.re0ega`;
+      await chai
+        .request(target.host())
+        .get(url)
+        .then((response) => {
+          expect(response).to.have.status(404);
+        }).catch((e) => {
+          e.message = `At ${target.host()}${url}\n      ${e.message}`;
+          throw e;
+        });
+    }).timeout(20000);
   });
 });

--- a/test/utils.js
+++ b/test/utils.js
@@ -53,11 +53,7 @@ function retrofit(fn) {
     return {
       statusCode: resp.status,
       body: String(resp.body),
-      headers: [...resp.headers.keys()].reduce((result, key) => {
-        // eslint-disable-next-line no-param-reassign
-        result[key] = resp.headers.get(key);
-        return result;
-      }, {}),
+      headers: Object.fromEntries(resp.headers),
     };
   };
 }

